File: backend/test_save.py
import requests
import json

def test_save_catalog():
    # 1. Login
    url = 'http://127.0.0.1:8000/api/auth/login/'
    data = {
        'username': 'bhavanbadhe@gmail.com',
        'email': 'bhavanbadhe@gmail.com',
        'password': 'bhavan@123'
    }
    response = requests.post(url, json=data)
    print("Login:", response.status_code)
    
    if response.status_code == 200:
        cookies = response.cookies
        
        # 2. Create Catalog
        cat_url = 'http://127.0.0.1:8000/api/catalogs/'
        cat_data = {
            "name": "Test Catalog",
            "settings": {}
        }
        # Include CSRF token if needed, dj-rest-auth cookie login doesn't enforce it by default here but we'll see
        cat_res = requests.post(cat_url, json=cat_data, cookies=cookies)
        print("Create catalog:", cat_res.status_code, cat_res.text)
        
        if cat_res.status_code == 201:
            catalog_id = cat_res.json()['id']
            
            # 3. Save Page
            page_url = f'http://127.0.0.1:8000/api/catalogs/{catalog_id}/save_page/'
            page_data = {
                "pageNumber": 1,
                "type": "interior",
                "elements": [],
                "categoryId": None
            }
            page_res = requests.post(page_url, json=page_data, cookies=cookies)
            print("Save page:", page_res.status_code, page_res.text)

if __name__ == "__main__":
    test_save_catalog()

File: backend/test_save.js
const http = require('http');

async function testSave() {
  try {
    const loginRes = await fetch('http://127.0.0.1:8000/api/auth/login/', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ email: 'bhavanbadhe@gmail.com', password: 'bhavan@123' })
    });
    
    console.log('Login Status:', loginRes.status);
    const cookies = loginRes.headers.get('set-cookie');
    console.log('Cookies:', cookies);
    
    if (loginRes.status === 200) {
      const createRes = await fetch('http://127.0.0.1:8000/api/catalogs/', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Cookie': cookies
        },
        body: JSON.stringify({ name: 'Test Catalog', settings: {} })
      });
      
      console.log('Create Catalog Status:', createRes.status);
      const createBody = await createRes.json();
      console.log('Create Catalog Body:', createBody);
      
      if (createRes.status === 201) {
        const catalogId = createBody.id;
        const savePageRes = await fetch(`http://127.0.0.1:8000/api/catalogs/${catalogId}/save_page/`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            'Cookie': cookies
          },
          body: JSON.stringify({
            pageNumber: 1,
            type: 'interior',
            elements: [],
            categoryId: null
          })
        });
        console.log('Save Page Status:', savePageRes.status);
        console.log('Save Page Body:', await savePageRes.text());
      }
    }
  } catch (err) {
    console.error('Error:', err);
  }
}

testSave();
